plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -65,7 +64,7 @@ func (l *lifecycle) Admit(a admission.Attributes) (err error) {
 		// this will cause a live lookup of the namespace to get its latest state even
 		// before the watch notification is received.
 		if a.GetOperation() == admission.Delete {
-			l.store.Delete(&api.Namespace{
+			l.informerFactory.Namespaces().Informer().GetStore().Delete(&api.Namespace{
```

Comment:
This still isn't safe.  How about reverting the change for this admission plugin and circle back to it separately.